src/TestInfrastructure/Extensions/FileSystemExtensions.cs

```diff
@@ -0,0 +1,63 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO.Abstractions;
+using Moq;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Extensions
+{
+    /// <summary>
+    /// Extension methods to simplify working with IFileSystem mocks
+    /// </summary>
+    public static class FileSystemExtensions
+    {
+        public static Mock<IFileSystem> FileDoesNotExist(this Mock<IFileSystem> fileSystem, string fullPath) =>
```

Comment:
We don't strictly need three "FileExists"-related methods - just the last one would do. However, I'm experimenting with whether the first two are easier to read/more declarative. If not, we can ditch them later.

Reply:
I think it's definitely easier to read, but I'd add "Set" prefix to them, otherwise it's a bit confusing if it's a verify or a setup.